packages/checkbox/index.js

```diff
@@ -0,0 +1,188 @@
+// The MIT License
+//
+// Copyright (c) 2018 Google, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the 'Software'), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED 'AS IS', WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import React, {Component} from 'react';
+import classnames from 'classnames';
+import PropTypes from 'prop-types';
+import {MDCCheckboxFoundation} from '@material/checkbox/dist/mdc.checkbox';
+import withRipple from '@material/react-ripple';
+
+import NativeControl from './NativeControl';
+
+export class Checkbox extends Component {
+
+  constructor(props) {
+    super(props);
+    this.inputElement_ = React.createRef();
+    this.foundation_ = null;
+    this.state = {
+      checked: props.checked,
+      indeterminate: props.indeterminate,
+      classList: new Set(),
+    };
+  }
+
+  componentDidMount() {
+    this.foundation_ = new MDCCheckboxFoundation(this.adapter);
+    this.foundation_.init();
+    this.foundation_.setDisabled(this.props.disabled);
+    // indeterminate property on checkboxes is not supported:
+    // https://github.com/facebook/react/issues/1798#issuecomment-333414857
+    if (this.inputElement_.current) {
+      this.inputElement_.current.indeterminate = this.state.indeterminate;
+    }
+  }
+
+  componentDidUpdate(prevProps) {
+    const {
+      checked,
+      indeterminate,
+      disabled,
+    } = this.props;
+
+    if (checked !== prevProps.checked) {
```

Comment:
this and the below can be put together in one setState call. In order to update indeterminate or checked, you will need to set both, so putting it together in one call is fine.

```
 if (checked !== prevProps.checked || indeterminate !== prevProps.indeterminate) {
  this.setState({indeterminate, checked})
... 
```

Then put the setState call into a function, since its shared logic from the onChange method